NUlid/Ulid.cs

```diff
@@ -635,35 +658,39 @@ public override bool Equals(object? obj)
     /// </returns>
     public int CompareTo(Ulid other)
     {
-        var d = other.ToByteArray();
-
-        return _a != d[0]
-            ? _a.CompareTo(d[0])
-            : _b != d[1]
-            ? _b.CompareTo(d[1])
-            : _c != d[2]
-            ? _c.CompareTo(d[2])
-            : _d != d[3]
-            ? _d.CompareTo(d[3])
-            : _e != d[4]
-            ? _e.CompareTo(d[4])
-            : _f != d[5]
-            ? _f.CompareTo(d[5])
-            : _g != d[6]
-            ? _g.CompareTo(d[6])
-            : _h != d[7]
-            ? _h.CompareTo(d[7])
-            : _i != d[8]
-            ? _i.CompareTo(d[8])
-            : _j != d[9]
-            ? _j.CompareTo(d[9])
-            : _k != d[10]
-            ? _k.CompareTo(d[10])
-            : _l != d[11]
-            ? _l.CompareTo(d[11])
-            : _m != d[12]
-            ? _m.CompareTo(d[12])
-            : _n != d[13] ? _n.CompareTo(d[13]) : _o != d[14] ? _o.CompareTo(d[14]) : _p != d[15] ? _p.CompareTo(d[15]) : 0;
+        return _a != other._a
```

Comment:
Is it possible to convert he Ulid to a `UInt128` in such a way that the ordering of UInt128 is the same as of the Ulids? Or instead of UInt128 any other type that fulfills the same ordering.
Thus the `CompareTo` boils down a simpler branching.

Or can `CompareTo` considered cold enough not worth further optimizations?